src/doom/tests/savegame_test.cpp

```diff
@@ -0,0 +1,181 @@
+#include "catch.hpp"
+#include <array>
+
+extern "C" {
+#include "doomtype.h"
+#include "p_saveg.h"
+
+byte  saveg_read8_in_stream   (FILE* stream, boolean* error);
+extern short saveg_read16_in_stream  (FILE* stream, boolean* error);
+extern int   saveg_read32_in_stream  (FILE* stream, boolean* error);
+extern void  saveg_write16_in_stream (FILE* stream, short value, boolean* error);
+extern void  saveg_write32_in_stream (FILE* stream, int value  , boolean* error);
+}
+
+static_assert(sizeof(boolean) == sizeof(byte));
+
+// ------------------ 8 BIT SCENARIOS ------------------
+// write
+SCENARIO("Writing 8 bits in a file", "[write]") {
+  GIVEN("An empty file")
+  {
+    std::array<byte, 2> buf{};
+    FILE *test_file = fmemopen(buf.data(), 2*sizeof(byte), "a");
+    setbuf(test_file, NULL);
+
+    AND_GIVEN("A value to write")
+    {
+      byte value = 0x43;
+
+      AND_GIVEN("No prior error")
+      {
+        boolean err = false;
```

Comment:
I think those two could be collapsed together
```suggestion
    AND_GIVEN("A value to write and no prior error")
    {
      byte value = 0x43;
      boolean err = false;
```